### DISCUSSION TITLE
Auto sync/processing concurrency limit

### DISCUSSION THREAD

**Discussion by mattops** [argoproj/argo-cd#10926] [Q&A] [answered]
Is there a way to limit the number of syncs that are occurring concurrently?

Reason for asking is that it is trivial to apply a code change in git that affects all apps running in the cluster.  This may trigger a rolling update of all pods. Depending on the rolling update settings the cluster may need significantly more resource available to accommodate the new pods that are being updated. 

There are probably plenty of other reasons why you would want to limit the rollout of big cluster wide changes too.

I'm not sure if there are controls in place already or not, I cannot find anything in the documentation - is there a way to limit the number of concurrent sync's of apps (or status progressing)?

I did [find this option](https://argo-cd.readthedocs.io/en/stable/operator-manual/server-commands/argocd-application-controller/) `--kubectl-parallelism-limit` but I don't think it will do what we need - I suspect the it will still throw out a lot of rolling update commands that may cause issues.

**Comment by jgwest** [comment, marked as answer]
Hi @mattops, this doc page is likely what you are looking for: https://argo-cd.readthedocs.io/en/stable/operator-manual/high_availability/#scaling-up

`--status-processors` and `--operation-processors` will control the number of concurrent update/syncs that are running, and `--parallelismlimit` controls how many manifests are generated at once (e.g. how many git clone/kustomize/helm processes are running at a time).
